seleniumJSEnv/src/main/scala/org/scalajs/jsenv/selenium/FileMaterializers.scala

```diff
@@ -1,41 +1,76 @@
 package org.scalajs.jsenv.selenium
 
+import scala.collection.JavaConverters._
+
 import java.io._
 import java.nio.file._
 import java.net._
 
-import org.scalajs.core.tools.io._
-import org.scalajs.jsenv.VirtualFileMaterializer
+import org.scalajs.io._
 
-private[selenium] trait FileMaterializer {
-  def materialize(vf: VirtualTextFile): URL
-}
+private[selenium] abstract class FileMaterializer(bypassRealFiles: Boolean) {
+  private val tmpSuffixRE = """[a-zA-Z0-9-_.]*$""".r
+
+  private[this] var tmpFiles: List[Path] = Nil
+
+  final def materialize(vf: VirtualBinaryFile): URL = vf match {
+    case vf: FileVirtualBinaryFile if bypassRealFiles => vf.file.toURI.toURL
```

Comment:
Instead of a boolean `bypassRealFiles`, I suggest that `materialize` not be `final` and not handle this `case` (i.e., it always behaves as if `!bypassRealFiles`. Then you can override `materialize` in `TempDirFileMaterializer` to take care of this case, and delegate to `super.materialize(vf)` when `vf` is not a `FileVirtualBinaryFile`.